client/state/sharing/services/selectors.js

```diff
@@ -9,6 +19,58 @@ export function getKeyringServices( state ) {
 }
 
 /**
+ * Returns an object of service objects with the specified type.
+ *
+ * @param  {Object} state Global state tree
+ * @param  {String} type  Type of service. 'publicize' or 'other'.
+ * @return {Array}        Keyring services, if known.
+ */
+export function getKeyringServicesByType( state, type ) {
+	return filter( getKeyringServices( state ), { type: type } );
```

Comment:
You can use object shorthand here: `{ type }`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#New_notations_in_ECMAScript_2015